### PR TITLE
Fix units hiding behind high walls

### LINKED ISSUE #1078: High walls block projectiles
Units behind top side of walls can't be directly damaged by missiles (potentially other attacks too) and units on that position have their projectiles blocked.

Screenshot for illustration:

Screenshot_20240726_173254

### PR DESCRIPTION
**Pull request by OcelotWalrus**
Fixes #1078.

### CODE REVIEW

**Comment by Dzierzan** [comment]
I do not approve this change. I'd rather find more proper solution than making passable terrain unpassable.

**Comment by OcelotWalrus** [comment]
> I do not approve this change. I'd rather find more proper solution than making passable terrain unpassable.

image

**Comment by Dzierzan** [comment]
Check the harbor map, you'll understand why this idea is not that great. Try to do something with hitbox instead.

**Comment by OcelotWalrus** [comment]
Sorry but I don't see why.

**Comment by Dzierzan** [comment]
South and north passages are 3 tile wide. If you that PR is merged, then west and east entries will be only 2 tiles wide.

image

**Comment by OcelotWalrus** [comment]
Oh yeah I see. I'll try to do something with the hitbox.

**Comment by OcelotWalrus** [comment]
@Mailaender the problem is that whether it has no hitbox, or a small one, it'll never look good with bullets firing on/in it. I don't think units should be able to go on these tiles. Maps like harbour that Dzierzan mentioned would then need to be adapted to that change.

**Comment by OcelotWalrus** [comment]
Actually @Mailaender, removing the hitbox doesn't cause weird visual bugs with bullets passing through, as these tiles are drawn over them anyway.

**Comment by Mailaender** [comment]
Still broken
grafik

File: mods/hv/rules/buildings.yaml [hunk 1]
```diff
@@ -2036,7 +2036,6 @@ VGATE:
 
 WALLT:
 	Inherits: ^Wall
-	Inherits@Shape: ^1x1Shape
 	Building:
 		Dimensions: 1,1
 		Footprint: =
```
Comment (Mailaender) [inline mods/hv/rules/buildings.yaml:2041]:
```suggestion
		Footprint: x
```
Just disallow units hiding there.

Comment (OcelotWalrus) [reply mods/hv/rules/buildings.yaml:2041]: Okay!

**Pull request merged**